packages/db/src/collection.ts

```diff
@@ -1538,8 +1603,11 @@ export class CollectionImpl<
     // Create a new transaction with a mutation function that calls the onUpdate handler
     const directOpTransaction = createTransaction<T>({
       mutationFn: async (params) => {
-        // Call the onUpdate handler with the transaction
-        return this.config.onUpdate!(params)
+        // Call the onUpdate handler with the transaction and collection
+        return this.config.onUpdate!({
+          ...params,
+          collection: this as unknown as Collection<T, TKey, TUtils>,
```

Comment:
it's handy to have direct access to the collection inside onInsert, etc. handlers